### DISCUSSION TITLE
Is using GuiDropdownBox with Camera2D supported?

### DISCUSSION THREAD

**Discussion by nomad-software** [raysan5/raylib#3129] [Q&A]
I've written the following test to draw a `GuiDropdownBox` and then move the `Camera2D` by pressing the right mouse button and moving the mouse.

When the mouse moves and the camera target changes, the hitbox for triggering the drop down box remains at pos 0,0 and so becomes disconnected from where the control is actually being drawn on screen. Is there anyway to use `Camera2D` with gui controls and have them work as expected?

This example uses Go bindings.

```go
package main

import (
	gui "github.com/gen2brain/raylib-go/raygui"
	rl "github.com/gen2brain/raylib-go/raylib"
)

func main() {
	rl.InitWindow(1024, 768, "Gui test")
	defer rl.CloseWindow()

	rl.SetWindowState(rl.FlagWindowResizable)

	var active int32 = 0
	var editMode bool

	target := rl.NewVector2(0, 0)
	cam := rl.NewCamera2D(rl.NewVector2(0, 0), target, 0, 1.0)

	for !rl.WindowShouldClose() {

		if rl.IsMouseButtonDown(rl.MouseRightButton) {
			cam.Target = rl.Vector2Negate(rl.GetMousePosition())
		}

		rl.BeginDrawing()
		rl.BeginMode2D(cam)
		rl.ClearBackground(rl.White)

		if editMode {
			gui.Lock()
		}

		if gui.DropdownBox(rl.NewRectangle(0, 0, 200, 50), "one;two;three", &active, editMode) {
			editMode = !editMode
		}

		gui.Unlock()

		rl.EndMode2D()
		rl.EndDrawing()
	}
}
```

**Comment by Peter0x44** [comment]
I took  a look at this, and saw GuiDropDownBox uses:
`2137         Vector2 mousePoint = GetMousePosition();`
This is the same for really all the raygui functions, there isn't a way to give a camera or scale
So, I think the answer is "no"

**Comment by raysan5** [comment]
I haven't tested it but I think you can use `SetMouseScale()` and `SetMouseOffset()` with Camera parameters to do it. Just note that camera rotation won't work.

**Comment by hexmaster111** [comment]
I used this in my application, Note that it dose not support zooming, but that can likely be added with the `SetMouseOffset()` like Raysan5 had said. 
```c#
BeginDrawing();
    ClearBackground(RAYWHITE);
    BeginMode2D(camera);
        SetMouseOffset((int)camera.target.X, (int)camera.target.Y);
            var mp = GetMousePosition();
            DrawCircle((int)mp.X, (int)mp.Y, 5, RED);
            if (GuiButton(new Rectangle(0, 0, 128, 24), "Some Button")) { }
        SetMouseOffset(0, 0);
    EndMode2D();

// HUD ---------------
    if (GuiButton(new Rectangle(0, 0, 128, 24), "Some Button")) { }
EndDrawing();
```
(note that this example is with the C# bindings)
Hope this helps!

**Comment by Z6dev** [comment]
I know I am very late... but if anyone in the future comes to the same problem as me, then here it is:  
  
You can account for zoom by using:
``` C
SetMouseOffset(   
			(camera.target.x * camera.zoom) - camera.offset.x,
			(camera.target.y * camera.zoom) - camera.offset.y  
);   
SetMouseScale(1.0f / camera.zoom, 1.0f / camera.zoom);  
```

and if you are planning to render a HUD after that, dont forget to reset:  
``` C
SetMouseOffset(0, 0); // Reset coords for HUD  
SetMouseScale(1.0f, 1.0f); // The scale too  
```
Hope this helps Someone!
